core/trino-main/src/main/java/io/trino/operator/WorkProcessorPipelineSourceOperator.java

```diff
@@ -332,7 +332,7 @@ private void updatePeakMemoryReservations(int operatorIndex)
 
                         succinctBytes(context.inputDataSize.get()),
                         context.inputPositions.get(),
-                        context.inputPositions.get() * context.inputPositions.get(),
+                        (double) context.inputPositions.get() * context.inputPositions.get(),
```

Comment:
nit:
commit message should be sth like `Prevent integer overflow when computing statistics`